### DISCUSSION TITLE
Model transformation

### DISCUSSION THREAD

**Discussion by phovorka** [google/model-viewer#3903] [Q&A]
WHERE:  Example-->Materials and Scenes-->Model transformation
DESCRIPTION OF PROBLEM:
++  you can change orientation (rotation) in degrees
++ you can change scale
--  what about position???
There is missing possibility to move with model(s) (materials)....

WHY MOVE IS IMPORTANT?
-  for Exploded view, parametrical input etc. 

Workaround for exploded view would be animation... but this is not a correct way...

**Comment by elalish** [comment]
That is [`camera-target`](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-cameraTarget), since movement of the camera isn't really discernible from movement of the scene.

**Reply by phovorka** [reply]
Disagree. Exploded view cannot be done with camera-target. Or how would you make somthing like this..
![image](https://user-images.githubusercontent.com/67637867/198024937-953706e1-5d8f-4762-8af4-37e38ce921c6.png)

**Reply by elalish** [reply]
Oh, I had assumed you were talking about transforming the whole scene since you said we support scale and orientation. We don't have any API for transforming individual meshes; we're intentionally higher-level than that. Yes, animation is the way to do exploded view in MV. If you want deeper hooks, you can fork us (or hack down to our $scene) and use the full power of the three.js game engine API.

**Reply by UnityJu** [reply]
@elalish How can I customize the model viewer to implement the model movement?
My goal is to develop a simple webAR app where users can place the car on the floor and drive it using controls(forward, backward, rotate). Not for each individual mesh.
Is it possible with the model viewer?
Thanks.

**Reply by elalish** [reply]
That might be possible with `camera-target` and `orientation`, though it'll probably feel pretty hacky. I'm working on a API to make this better (and support multiple models).

**Reply by UnityJu** [reply]
@elalish  So is it not possible to place the car and drive it using UI control buttons? like forward and backward, rotate?
I want to move the model itself. not camera.

**Reply by elalish** [reply]
I believe it may be as I stated above, but you'll need to code it up and give it a try. Have you attempted? `camera-target` currently acts like a position offset in WebXR mode as I recall.

**Reply by UnityJu** [reply]
Okay, I will give it a try. 
Please keep me posted if your team adds the new feature of model-target movement and rotation, etc.
Thanks.
